src/applications/vaos/appointment-list/redux/actions.js

```diff
@@ -423,49 +424,100 @@ export function fetchPastAppointments(startDate, endDate, selectedIndex) {
 
 export function fetchRequestDetails(id) {
   return async (dispatch, getState) => {
-    const state = getState();
-    const { appointmentDetails, requestMessages, pending } = state.appointments;
-    const request = appointmentDetails[id] || pending?.find(p => p.id === id);
+    try {
+      const state = getState();
+      let request = selectRequestById(state, id);
+      let facilityId = getVAAppointmentLocationId(request);
+      let facility = state.appointments.facilityData?.[facilityId];
+
+      if (!request || (facilityId && !facility)) {
```

Comment:
I don't want to flash a loading spinner if we don't have to wait for the request/appointment or facility data